python/pyspark/sql/functions.py

```diff
@@ -2129,7 +2129,8 @@ def _create_udf(f, returnType, vectorized):
     def _udf(f, returnType=StringType(), vectorized=vectorized):
         if vectorized:
             import inspect
-            if len(inspect.getargspec(f).args) == 0:
+            argspec = inspect.getargspec(f)
+            if len(argspec.args) == 0 and argspec.varargs is None:
```

Comment:
BTW, let's address [this comment](https://issues.apache.org/jira/browse/SPARK-21190?focusedCommentId=16185166&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16185166) while we are here.

Reply:
Fixed